### PR TITLE
Code coverage and bug fixes for bcelifier

### PR DESCRIPTION
**Pull request by nbauma109**
While doing coverage, I found a few bugs that needed to be fixed. I collected auxiliary jacoco execution reports for the exec(...) calls and merged them into the main one.

### CODE REVIEW

**Comment by garydgregory** [comment]
Hello @nbauma109 

Thank you for your PR. There is a lot here but it feels like a mish-mash of different more or less related items.

For example, the PR adds the constructor `org.apache.bcel.generic.PUSH.PUSH(ConstantPoolGen, ArrayType)` but it is neither used nor tested. Either remove it, write a test, or use it and write a test for it. If there are missing APIs, I suggest you cover those in separate PRs. Mixing this new code here makes this PR harder to review.

Based on the title of the PR, I was expecting only tests, it's ok to have other work in here, the different pieces should be clearly documented in the PR description.

**Comment by nbauma109** [comment]
The constructors `org.apache.bcel.generic.PUSH.PUSH(ConstantPoolGen, ArrayType)` and `org.apache.bcel.generic.PUSH.PUSH(ConstantPoolGen, ObjectType)` are not called directly by the code of the project but the BCELifier code generator creates calls to it. In target directory, a source file `Java8Example2Creator.java` is created with a call to it 

`il.append(new PUSH(_cp, new ArrayType(Type.STRING, 1)));`

I generated a separate jacoco exec file for the execution of this code, which is merged into the main jacoco.exec, and coverage report is online:

https://codecov.io/gh/apache/commons-bcel/pull/171/diff?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=The+Apache+Software+Foundation#diff-c3JjL21haW4vamF2YS9vcmcvYXBhY2hlL2JjZWwvZ2VuZXJpYy9QVVNILmphdmE=

File: src/main/java/org/apache/bcel/util/BCELFactory.java [hunk 1]
```diff
@@ -299,7 +323,7 @@ public void visitLocalVariableInstruction(final LocalVariableInstruction i) {
 
     @Override
     public void visitRET(final RET i) {
-        printWriter.println("il.append(new RET(" + i.getIndex() + ")));");
+        printWriter.println("il.append(new RET(" + i.getIndex() + "));");
```
Comment (garydgregory) [inline src/main/java/org/apache/bcel/util/BCELFactory.java:326]: Good catch!

**Comment by garydgregory** [comment]
@nbauma109 
Thank you for your reply. Merged.

**Pull request merged**